src/ResourceManager.cpp

```diff
@@ -403,8 +402,7 @@ Graphics::Shader* ResourceManager::shader(const string& filename)
         return _shaders.at(filename).get();
     }
 
-    Graphics::Shader* shader = nullptr;
-    shader = new Graphics::Shader(filename);
+    Graphics::Shader* shader = new Graphics::Shader(filename);
 
     _shaders.insert(make_pair(filename, unique_ptr<Graphics::Shader>(shader)));
```

Comment:
It seems that we can use emplace() method like
_shaders.emplace(filename, make_unique<Graphics::Shader>(filename)));